NEXUS-5192: Implementing the property to control "local avail check"

cstamas opened the pull request:
Making it default to false (reverse behaviour as so far), but still
making it possible to be enabled if needed.

This change reverts how NX behaved so far.

CI:
https://builds.sonatype.org/view/nexus-features/job/nexus-oss-its-feature/423/

cstamas committed:
NEXUS-5192: Implementing the property to control "local avail check"

Making it default to false (reverse behaviour as so far), but still
making it possible to be enabled if needed.

cstamas commented:
CI +1'd
(but both slave jobs failed as CompositeExceptionTest from master was missing license header, fixed in b4bfb187ac48e0a2fb0e3856c3e19f7a0f5764bb)

ifedorenko commented:
+1 

code format looks proper in the diff

nabcos commented on nexus/nexus-core-plugins/nexus-indexer-lucene-plugin-parent/nexus-indexer-lucene-plugin/src/main/java/org/sonatype/nexus/index/treeview/DefaultMergedTreeNodeFactory.java:
```diff
@@ -159,18 +118,65 @@ protected ResourceStoreRequest getResourceStoreRequest( String path )
+        }
+        catch ( Exception e )
+        {
+            // TODO: do something here?
```
at least some logging?

adreghiciu commented:
+1

nabcos commented:
+1 but maybe handle the TODO? It's only for UI though, so ignoring it may be fine, but then the comment should say so.

cstamas committed:
NEXUS-5192: Elaborating the TODO section

Providing meaningful messages in case of some exceptions.

As this code predates SISU and SLF4J change, and is NOT a component,
logging with vanilla plexus was not quite possible from it without
some trickery (like passing around Plexus Logger instance). Fixing
it.
